Export aggregation difference to excel (#237)
IAMconsortium/nomenclature: doc/source/user_guide/model-mapping.rst

   *region_b*. The **provided results** take **precedence** over the aggregated ones.
   Additionally, the aggregation is computed and compared to the provided results. If
   there are discrepancies, a warning is written to the logs.

   .. note::

      Please note that in case of differences no error is raised. Therefore it is
      necessary to check the logs to find out if there were any differences. This is
      intentional since some differences might be expected.

> How to get aggregation differences locally
> ------------------------------------------

Reviewer: ```suggestion
Computing differences between original and aggregated data
----------------------------------------------------------
```

> In order to get the full differences perform the following steps:

Reviewer: ```suggestion
In order to get the differences between the original data (e.g., results reported by the model)
and the data aggregated according to the region mapping, perform the following steps:
```

> 1. Make sure you have `pyam-iamc >= 1.7.0`, `nomenclature-iamc>=0.9.2` and

Author: Sure let's make it 0.10 :smile:

`pandas >= 1.5.2` installed.
2. Clone the workflow directory of your project
3. Navigate to the workflow directory
4. Using a jupyter notebook or python script run the following:

.. code:: python

  from pyam import IamDataFrame
  from nomenclature import DataStructureDefinition, RegionProcessor

  data = IamDataFrame("/path/to/your/input/data.xlsx")

  dsd = DataStructureDefinition("definitions")
  processor = RegionProcessor.from_directory("mappings", dsd)

  # get the differences as a pandas dataframe
  # the value for the relative tolerances can be adjusted, defaults to 0.01

>   differences = processor.check_region_aggregation(data, rtol_difference=0.01)

Reviewer: Shouldn’t this return a tuple?

Author: The way that I've implemented it for now is that it only returns the differences. I could of course have it return both the result and the differences

Reviewer: I think returning both is more useful - otherwise, you have to run the computation twice if you want to have both the aggregated dataset *and* the differences (or you go to the internal method, which one should avoid).

Author: Sounds good, updated

>   differences = processor.check_region_aggregation(data, rtol_difference=0.01)

Reviewer: ```suggestion
  processed_data, differences = processor.check_region_aggregation(data, rtol_difference=0.01)
```

  differences.to_excel("differences.xlsx")

> For details on the check_region_aggregation function please refer to

Reviewer: ```suggestion
For details on this feature please refer to
```
